Bottom: 0b6425c4eb2400fcd4a766eb6e1d20493bc62966
Top:    d53928cfdc86969e4c2490b3574d823ac1749e5b
Author: Alexey Lyashkov <shadow@Alexeys-MacBook-Pro.local>
Date:   2016-01-15 13:23:55 +0300

Refresh of srv.patch

---

diff --git a/ib-sock.c b/ib-sock.c
index fa47955..8b9565f 100644
--- a/ib-sock.c
+++ b/ib-sock.c
@@ -201,7 +201,7 @@ void ib_socket_disconnect(struct IB_SOCK *sock)
 		printk("Failed to disconnect, conn: 0x%p err %d\n",
 			 sock,err);
 }
-
+/*****************************************************************************************/
 static unsigned long __take_event(struct IB_SOCK *sock, unsigned long *e)
 {
 	unsigned long events;
@@ -224,3 +224,43 @@ unsigned long ib_socket_poll(struct IB_SOCK *sock)
 
 	return mask;
 }
+
+/*****************************************************************************************/
+int ib_socket_bind(struct IB_SOCK *sock, uint32_t addr, unsigned port)
+{
+	struct sockaddr_in  sin;
+	int ret;
+
+	sin.sin_family = AF_INET,
+	sin.sin_addr.s_addr = (__force u32)htonl(addr);
+	sin.sin_port = (__force u16)htons(port);
+
+	ret = rdma_bind_addr(sock->is_id, (struct sockaddr *)&sin);
+	if (ret) {
+		printk(KERN_ERR "RDMA: failed to setup listener, "
+		       "rdma_bind_addr() returned %d\n", ret);
+		goto out;
+	}
+
+	ret = rdma_listen(sock->is_id, IB_LISTEN_QUEUE);
+	if (ret) {
+		printk(KERN_ERR "RDMA: failed to setup listener, "
+		       "rdma_listen() returned %d\n", ret);
+		goto out;
+	}
+	/* HELLO will done via CM (mad) packets */
+out:
+	return ret;
+
+}
+
+struct IB_SOCK *ib_socket_accept(struct IB_SOCK *parent)
+{
+	struct IB_SOCK *sock;
+
+	sock = list_first_entry_or_null(&parent->is_child, struct IB_SOCK, is_child);
+	if (sock) {
+		list_del_init(&sock->is_child);
+	}
+	return sock;
+}
diff --git a/ib-sock.h b/ib-sock.h
index 533e991..0a2f9a9 100644
--- a/ib-sock.h
+++ b/ib-sock.h
@@ -25,4 +25,8 @@ void ib_socket_disconnect(struct IB_SOCK *sock);
 /* sleep to wait poll event from socket */
 unsigned long ib_socket_poll(struct IB_SOCK *sock);
 
+int ib_socket_bind(struct IB_SOCK *sock, uint32_t addr, unsigned port);
+struct IB_SOCK *ib_socket_accept(struct IB_SOCK *parent);
+
+
 #endif
\ No newline at end of file
